central/vulnerabilityrequest/datastore/datastore_impl.go

```diff
@@ -231,6 +231,92 @@ func (ds *datastoreImpl) UpdateRequestExpiry(ctx context.Context, id, message st
 	return req, nil
 }
 
+func (ds *datastoreImpl) UpdateRequest(ctx context.Context, id string, update *common.UpdateRequest) (*storage.VulnerabilityRequest, error) {
+	if !features.UnifiedCVEDeferral.Enabled() {
+		return nil, errors.Errorf("set the environment variable %s=true and retry", features.UnifiedCVEDeferral.EnvVar())
+	}
+
+	// Only those with WRITE on either VulnerabilityManagementApprovals or VulnerabilityManagementRequests can make this change
+	if ok, err := requesterOrApproverSAC.WriteAllowedToAny(ctx); err != nil {
+		return nil, err
+	} else if !ok {
+		return nil, sac.ErrResourceAccessDenied
+	}
+
+	req, found, err := ds.store.Get(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+	if !found {
+		return nil, errors.Errorf("vulnerability request %s not found", id)
+	}
+
+	// If the user has VulnerabilityManagementRequests but not VulnerabilityManagementApprovals, then this can only be done if current user is the requester
+	if ok, err := requesterSAC.WriteAllowed(ctx); err == nil && ok {
+		if ok, err := approverSAC.WriteAllowed(ctx); err != nil || !ok {
+			user, err := authn.IdentityFromContext(ctx)
+			if err != nil {
+				return nil, err
+			}
+			if req.Requestor.Id != user.UID() {
+				return nil, errors.Wrap(sac.ErrResourceAccessDenied, "requests can only be updated by the original requester")
+			}
+		}
+	}
+
+	if req.GetExpired() || req.GetStatus() == storage.RequestStatus_DENIED {
```

Comment:
nit:  I would move this above the preceding SAC checks.  If the requests is expired or denied there is no point in doing that SAC check.

Reply:
Actually no. We do not want to leak information about the request to the users if they do not have the required permission to complete the operation. Note that `req` is pulled out of the database to which the user may or may not have permission.